website/src/content/deployment-examples/azure-static-web-apps.mdx

```diff
@@ -0,0 +1,116 @@
+- A Custom Application [configured in the Merchant Center](https://docs.commercetools.com/merchant-center/managing-custom-applications).
+
+# Configuration
+
+In your Custom Application config, provide the [Custom Application ID](/api-reference/application-config#envproductionapplicationid) you got when you configured the Custom Application in the Merchant Center.
+
+Moreover, you need to provide the [production URL](/api-reference/application-config#envproductionurl) from your Azure project. You can keep the standard Azure URL `https://<project>.1.azurestaticapps.net` or provide your custom domain.
+
+```json title="custom-application-config.json"
+{
+  "env": {
+    "production": {
+      "applicationId": "ckvtahxl90097sys6har1e6n3",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+## Using environment variables
+
+In case you want to avoid hardcoding certain values, for example the Application ID, or the Project key, you can [use variable placeholders](/api-reference/application-config#using-variable-placeholders) in your Custom Application config.
+
+```json title="custom-application-config.json" highlightLines="4"
+{
+  "env": {
+    "production": {
+      "applicationId": "${env:APPLICATION_ID}",
+      "url": "https://<project>.1.azurestaticapps.net"
+    }
+  }
+}
+```
+
+# Connect Azure with GitHub Actions
+
+The easiest way to deploy to Azure is to use [Static Web Apps](https://azure.microsoft.com/en-us/services/app-service/static/#overview) deployment service. This service enables first-class GitHub integration.
+
+Follow the steps in the [Azure Static Web App creator](https://portal.azure.com/#create/Microsoft.StaticApp) to create a new project and select `GitHub` as the deployment source.
+
+<Info>
+
+Make sure to grant Azure access to your repository in the following step.
+
+</Info>
+
+## Configuring build settings
+
+In the Static Web App setup process you need to configure the following things:
+
+* Select your organization, repository and branch.
+* In the **Build details** section select `Custom` as the build preset.
+* Provide the following build settings:
+    * App location: `/`
+    * Output location: `public`
+
+<Info>
+
+If your Custom Application config [requires environment variables](#using-environment-variables), make sure to provide them to your GitHub repository secrets and update the [GitHub Action workflow file](https://docs.microsoft.com/en-us/azure/static-web-apps/build-configuration?tabs=github-actions#environment-variables) accordingly.
+
+For instance, when using `APPLICATION_ID` environment variable in your Custom Application, add `APPLICATION_ID` repository secret in GitHub settings and edit the GitHub Action workflow file:
+
+```yaml highlightLines="9-10"
+...
+
+with:
+  ...
+  action: "upload"
+  app_location: "/"
+  api_location: ""
+  output_location: "public"
+env:
+  APPLICATION_ID: ${{ secrets.APPLICATION_ID }}
+```
```

Comment:
Let's use a more "complete" example that includes the github action itself

````suggestion
See example below for defining environment variables for the GitHub action:

```yaml title="deployment.yml" highlightLines="9-10"
- name: Build And Deploy
  uses: Azure/static-web-apps-deploy@v1
  with:
    action: "upload"
    app_location: "/"
    api_location: ""
    output_location: "public"
  env:
    APPLICATION_ID: ${{ secrets.APPLICATION_ID }}
    CLOUD_IDENTIFIER: gcp-eu
```
````